src/test/java/org/symphonyoss/symphony/messageml/elements/DialogTest.java

```diff
@@ -355,6 +355,151 @@ public void testMarkdownConversionWithoutFooter() throws Exception {
         context.getMarkdown());
   }
 
+  @Test
+  public void testWithForm() throws Exception {
+    String dialogId = "id-dialog";
+    String formId = "id-form";
+    String title = "title";
+    String body = "body";
+    String footer = "footer";
+    String input = "<messageML>"
+        + "<dialog id=\"" + dialogId + "\">"
+        + "<form id=\"" + formId + "\">"
+        + "<title>" + title + "</title>"
+        + "<body>" + body + "</body>"
+        + "<footer>" + footer + "</footer>"
+        + "</form>"
+        + "</dialog>"
+        + "</messageML>";
+    context.parseMessageML(input, null, MessageML.MESSAGEML_VERSION);
+    assertFormInDialogBuilt(context.getMessageML(), dialogId, Dialog.MEDIUM_WIDTH, Dialog.CLOSE_STATE, title, body, footer);
+  }
+
+  @Test
+  public void testWithValidElementInForm() throws InvalidInputException, IOException, ProcessingException {
+    String dialogId = "id-dialog";
+    String formId = "id-form";
+    String title = "title";
+    String body = "body";
+    String footer = "footer";
+    String input = "<messageML>"
+        + "<dialog id=\"" + dialogId + "\">"
+        + "<form id=\"" + formId + "\">"
+        + "<title>" + title + "</title>"
+        + "<body>"
+        + "<checkbox name=\"fruits\" value=\"body\">" + body + "</checkbox>"
+        + "</body>"
+        + "<footer>" + footer + "</footer>"
+        + "</form>"
+        + "</dialog>"
+        + "</messageML>";
+    final String expectedPattern = "^<div data-format=\"PresentationML\" data-version=\"2.0\">"
+        + "<dialog data-width=\"medium\" data-state=\"close\" id=\"\\S+-id-dialog\" open=\"\">"
+        + "<form id=\"id-form\"><div class=\"dialog-title\">title</div>"
+        + "<div class=\"dialog-body\">"
+        + "<div class=\"checkbox-group\"><input type=\"checkbox\" name=\"fruits\" value=\"body\" id=\"checkbox-group-\\S+\"/>"
+        + "<label for=\"checkbox-group-\\S+\">body</label></div></div>"
+        + "<div class=\"dialog-footer\">footer</div>"
+        + "</form>"
+        + "</dialog>"
+        + "</div>$";
+
+    context.parseMessageML(input, null, MessageML.MESSAGEML_VERSION);
+    assertFormInDialogBuilt(context.getMessageML(), dialogId, Dialog.MEDIUM_WIDTH, Dialog.CLOSE_STATE, title, body, footer);
+
+    final String presentationML = context.getPresentationML();
+    assertTrue(presentationML.matches(expectedPattern));
+  }
+
+  @Test
+  public void testWithInvalidElementInForm() {
```

Comment:
It seems one of the scenarios is missing. Interactive elements are not allowed if the dialog doesn't contain the form tag. But non-interactive elements like a card or an expandable-card are allowed. (See https://perzoinc.atlassian.net/wiki/spaces/DevX/pages/1986267475/Element+Dialog+-+MessageML+PresentationML+Form+Reply+Content+Export#Children-tags-of-the-Dialog)
```
<dialog id='uniqueID'>
    <title>A title</title>
    <body>
      <span>Something</span>
      <card>A card is allowed</card> <======================== VALID
      <button>A button</button> <============================= !INVALID
      <expandable-card>An expandable-card</expandable-card> <= VALID
    </body>
</dialog>
```

(I've just realized that we should also allow in all cases the button with type='cancel', because it will close the Dialog)

Reply:
Seems this is already tested in `testDialogWithNonInteractiveElementInTitle`, `...InBody`, `...InFooter`. The tests were already present before this PR.

From the conversation we had with Thibault, there is no need for the cancel button.